spec/unit/agent_spec.rb

```diff
@@ -236,6 +236,31 @@ def controlled_run(&block)
         expect(agent.should_fork).to be_falsey
       end
     end
+
+    describe 'when runtimeout is set' do
+      before(:each) do
+        Puppet[:runtimeout] = 1
+      end
+
+      it 'times out when a run exceeds the set limit' do
+        client = AgentTestClient.new
+        client.instance_eval do
+          def run(client_options = {})
+            # Simulate a hanging agent operation that also traps errors.
+            begin
+              ::Kernel.sleep(5)
+            rescue
+              nil
+            end
+          end
+        end
+
+        AgentTestClient.expects(:new).returns client
+        @agent.expects(:handle_runtimeout)
+
+        @agent.run
```

Comment:
The return value of run is important. We might want to also test for it?

Reply:
Good catch, I'll add an expectation for that.